src/tfr_navigation/src/navigation_goal_manager.cpp

```diff
@@ -19,37 +20,41 @@ NavigationGoalManager::NavigationGoalManager(const GeometryConstraints &c,
         ROS_WARN("    finish_line: %f",
                 constraints.get_finish_line());
     }
-    
+    ROS_INFO("frame: %s", reference_frame.c_str());
 }
 
-NavigationGoalManager::NavigationGoalManager(const GeometryConstraints &c):
-    NavigationGoalManager{c, tfr_msgs::NavigationGoal::UNSET} {}
-
-
 /**
  *  Initializes the goal based on the location code, flashes a warning if it's
  *  not recognized.
  * */
-move_base_msgs::MoveBaseGoal NavigationGoalManager::initialize_goal() {
+move_base_msgs::MoveBaseGoal NavigationGoalManager::initialize_goal(
+        tfr_utilities::LocationCode new_goal) {
+    goal = new_goal;
+    //set reference frame
+    nav_goal.target_pose.header.frame_id = reference_frame;
     nav_goal.target_pose.header.stamp = ros::Time::now();
-    //TODO integrate the bin reference frame here, upon completion of
-    //the bin transform publisher
-    switch(location_code)
+    //set translation goal
+    switch(goal)
     {
-        case(tfr_msgs::NavigationGoal::TO_MINING):
+        case(tfr_utilities::LocationCode::MINING):
             nav_goal.target_pose.pose.position.x =
                 constraints.get_safe_mining_distance();
-            //TODO 
             break;
-        case(tfr_msgs::NavigationGoal::TO_DUMPING):
+        case(tfr_utilities::LocationCode::DUMPING):
             nav_goal.target_pose.pose.position.x = constraints.get_finish_line();
             break;
-        case(tfr_msgs::NavigationGoal::UNSET):
+        case(tfr_utilities::LocationCode::UNSET):
             //leave it alone
             break;
         default:
-            ROS_WARN("location_code %d not recognized", location_code);
+            ROS_WARN("location_code %u not recognized",
+                    static_cast<uint8_t>(goal));
     }
+    //set relative rotation (none)
+    nav_goal.target_pose.pose.orientation.x = 0;
+    nav_goal.target_pose.pose.orientation.y = 0;
+    nav_goal.target_pose.pose.orientation.z = 0;
+    nav_goal.target_pose.pose.orientation.w = 1;
```

Comment:
What is "relative rotation" here, and why is it always 0?

Reply:
(0,0,0,1) means no rotation. 
![image](https://user-images.githubusercontent.com/15841243/35772666-d429ed3e-08f7-11e8-97f7-839b35957c91.png)

Reply:
This site is also really handy for making quaternions human readable

http://quaternions.online/

Reply:
the relevant frame is "bin_footprint" which is a point in the dead center of the bin, at ground level.

Reply:
So the relative rotation is relative to the orientation of the bin?

Reply:
yes.